agents/services/agents.js

```diff
@@ -8,24 +8,40 @@ const hooks = {
     create: [
       (hook) => {
         if (hook.result.type === 'person') {
+          const agentId = hook.result.id
+          var agentEmail
+
           return hook.app.service('taskPlans').create({
-            assigneeId: hook.result.id,
+            assigneeId: agentId,
             taskRecipeId: 'createProfile',
             params: {}
           })
           .then(() => {
-            return hook.app.service('credentials').get(hook.result.id)
+            return hook.app.service('credentials').find({ query: { agentId } })
+          })
+          .then(([cred]) => {
+            agentEmail = cred.email
+            return hook.app.service('tokens').create({
+              agentId,
+              service: 'credentials',
+              method: 'patch',
+              params: { serviceId: cred.id },
+              createdAt: new Date().toString()
+            })
           })
-          .then((cred) => {
+          .then((token) => {
             // TODO: IK: looks like the email isn't getting attached to the credential correctly, investigate dogstack-agents
             return hook.app.service('mailer').create({
               from: 'hello@cobuy.nz',
-              to: cred.email || 'no@email.com',
-              subject: 'Test Cobuy SignUp Email',
-              html: 'test email'
+              to: agentEmail || 'no@email.com',
+              subject: `You're invited to join Cobuy!`,
+              html: `
+                Hi -name-. You've been invited by -admin- to join -group- on Cobuy! Click <a href=http://localhost:3000/invited/${token.jwt}>here</a> to set your password and start co-buying!
```

Comment:
um, we can't hardcode the development url. keen for you to be able to access the base url at `app.get('url')`

Reply:
depends on the relevant code from https://github.com/root-systems/cobuy/commit/fd64c4bfe53a71203a2730158b22123af9e09317